backend/flask/app.py

```diff
@@ -124,20 +135,67 @@ def create_link_token(userId):
         return jsonify(json.loads(e.body)), e.status
 
 
-@app.route('/exchange_public_token', methods=['POST'])
-def exchange_public_token():
-    data = request.get_json()
-    public_token = data.get('public_token')
-    if not public_token:
-        return jsonify({'error': 'Public token is required'}), 400
-
+@app.route('/api/exchange_public_token/<userId>', methods=['POST'])
+def exchange_public_token(userId):
+    public_token = request.json['public_token']
     try:
-        exchange_response = plaid_client.item_public_token_exchange(public_token)
+        exchange_request = ItemPublicTokenExchangeRequest(
+            public_token=public_token
+        )
+        exchange_response = plaid_client.item_public_token_exchange(exchange_request)
         access_token = exchange_response['access_token']
-        return jsonify({'access_token': access_token}), 200
-    except plaid.ApiException as e:
-        return jsonify(json.loads(e.body)), e.status 
-    
+        item_id = exchange_response['item_id']
+        user = User.query.get(userId)
+        user.token = access_token # store the user's access token in user table
+        db.session.commit()
+        
+        return jsonify({
+            'access_token': access_token,
+            'item_id': item_id,
+            'public_token_exchange': 'complete'
+        })
+    except Exception as e:
+        return jsonify({'error': str(e)}), 500
+
+
+@app.route('/api/transactions/sync/<userId>', methods=['GET'])
+def transactions_sync(userId):
+    try:
+        # get the user's access_token from the user table
+        user = User.query.get(userId)
+        request = TransactionsSyncRequest(
+            access_token=user.token
+        )
+        transactions = plaid_client.transactions_sync(request)
+        # Transaction object is by default non-serialable hence why we transform it to a dictionary
+        transactions_data = {
+            "transactions": [transaction.to_dict() for transaction in transactions.added]
+        }
+        return clean_transaction_data(transactions_data)
+    except Exception as e:
+        return jsonify({'error': str(e)}), 500
+
+
+"""
+helper function to filter and simplify the JSON returned by the 
+[transactions/sync] endpoint into the relevant fields of
+amount, date, currency, primary, and detailed categories. 
+"""
+def clean_transaction_data(transaction_json):
+    cleaned_data = []
+    for transaction in transaction_json['transactions']:
+        # filter out transactions where money is deposited to account since we're concerned with expenditures
+        if(transaction["amount"] > 0):
+            clean_transaction = {
+                "amount": transaction["amount"],
```

Comment:
Is it ok if transaction doesn't contain "amount" (or any other properties below)? That means `clean_transaction` could have some null properties.

Reply:
I made a note to add another guard here in case of this– I'm going to make a PR dedicated to all these additional fixes immediately following the AG-Grid PR